cfme/tests/cli/test_appliance_update.py

```diff
@@ -26,27 +31,37 @@
 
 def pytest_generate_tests(metafunc):
     """The following lines generate appliance versions based from the current build.
-    Appliance version is split and minor_build is picked out for generating each version
+    Appliance version is split and z-version is picked out for generating each version
     and appending it to the empty versions list"""
-    versions = []
     version = find_appliance(metafunc).version
+    versions = []
 
-    split_ver = str(version).split(".")
-    try:
-        minor_build = split_ver[2]
-        assert int(minor_build) != 0
-    except IndexError:
-        logger.exception('Caught IndexError generating for test_appliance_update, skipping')
-    except AssertionError:
-        logger.debug('Caught AssertionError: No previous z-stream version to update from')
-        versions.append(
-            pytest.param(
-                "bad:{!r}".format(version),
-                marks=pytest.mark.uncollect('Could not parse minor_build from: {}'.format(version))
-            )
-        )
-    except Exception:  # diaper just in case
-        logger.exception('Exception hit parsing version for test_appliance_update')
+    old_version_pytest_arg = metafunc.config.getoption('old_version')
+    if old_version_pytest_arg == 'same':
+        versions.append(version)
+    elif old_version_pytest_arg is None:
+        split_ver = str(version).split(".")
+        try:
+            z_version = int(split_ver[2])
+        except (IndexError, ValueError) as e:
+            logger.exception("Couldn't parse version: %s, skipping", e)
+            versions.append(pytest.param("bad:{!r}".format(version), marks=pytest.mark.uncollect(
+                reason='Could not parse z_version from: {}'.format(version)
+            )))
+        else:
+            if z_version < 1:
```

Comment:
could it be -1? :)

Reply:
You never know... (:

Reply:
Yeah, the current code used LYBL. I am changing it to be more EAFP - like. https://docs.python.org/3.4/glossary.html#term-lbyl